app/src/main/java/com/swent/assos/model/view/NewsViewModel.kt

```diff
@@ -0,0 +1,31 @@
+package com.swent.assos.model.view
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.swent.assos.model.data.Association
+import com.swent.assos.model.data.News
+import com.swent.assos.model.service.DbService
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class NewsViewModel @Inject constructor(private val dbService: DbService) : ViewModel() {
+  private val _allNews = MutableStateFlow(emptyList<News>())
+  val allNews = _allNews.asStateFlow()
+
+  init {
+    viewModelScope.launch(Dispatchers.IO) { dbService.getAllNews().let { _allNews.value = it } }
```

Comment:
avoid hard coded dispatchers, allow a possible injection as a parameter (with a default)

Reply:
Done thank you !